test/unit/math/rev/mat/prob/bernoulli_logit_glm_lpmf_test.cpp

```diff
@@ -378,3 +378,31 @@ TEST(ProbDistributionsPoissonLogGLM,
   EXPECT_THROW(stan::math::bernoulli_logit_glm_lpmf(y, x, alpha, betaw2),
                std::domain_error);
 }
+
+TEST(ProbDistributionsBernoulliLogitGLM, test_scalar_rowvec_stdvec) {
+  Matrix<double, Dynamic, Dynamic> x(3, 2);
+  x << -12, 46, -42, 24, 25, 27;
+  Matrix<double, Dynamic, 1> beta(2, 1);
+  beta << 0.3, 2;
+  double alpha = 0.3;
+
+  std::vector<double> y_vec(3);
+  y_vec[0] = 0;
+  y_vec[1] = 0;
+  y_vec[2] = 0;
+  EXPECT_NO_THROW(stan::math::bernoulli_logit_glm_lpmf(y_vec, x, alpha, beta));
+
+  Matrix<double, 1, Dynamic> y_rvec(1, 3);
+  y_rvec << 1, 0, 1;
+  EXPECT_NO_THROW(stan::math::bernoulli_logit_glm_lpmf(y_rvec, x, alpha, beta));
+
```

Comment:
current implementation only allows scalar alpha (it's a global intercept). It would be cool to allow varying intercept models for these GLM's by setting vector intercepts but for now I'm just going to do different types for beta.

Reply:
You are wrong. Vector alpha is allowed (at least on C++ side - maybe not in stan language). It is also in the tests. You can search any GLM test source for "alphavec" - that is how variable is named when testing it.

Reply:
yeah, you're correct. I'm thinking from a modeling perspective. I'm going to go ahead and leave the vector tests for the intercept out. It brings up questions such as: varying intercept for all observations, only some observations, and it would require additional programming and discussion to make sure we're allowing user flexibility and we're making the C++ code robust to invalid user input.

There are also other ways currently in the language that allow use multilevel models so it's not a priority to add vector length scale at this time.

I'm also being vague by using the word "observations". One concrete and well known example are demographic covariates.

Reply:
That discoussion would be necessary if we were discussing how to implement varying intercept. Since it is already implemented (options are one for all observations and one per observation) I think there is no reason not to test it. If you still disagree feel free to leave it out and then I will probably write the tests for it.

Reply:
So above I was referring to specific column/field/features. If we use vector alpha we're using separate intercepts for each row (could be each patient, etc). I'll just document accordingly so it's clear to users what model it's specifying.

Thanks, and sorry for being abrasive. I'll go ahead and add this feature

Reply:
I've added a test for std::vector or Eigen's row vector or column vector and it throws an error. I'm skipping this for now.

Reply:
Which argument it throws for? I think it should work for all ... I guess I will have to take a look.